pkg/apis/junit/types.go

```diff
@@ -0,0 +1,110 @@
+package junit
+
+import (
+	"encoding/xml"
+)
+
+// The below types are directly marshalled into XML. The types correspond to jUnit
+// XML schema, but do not contain all valid fields. For instance, the class name
+// field for test cases is omitted, as this concept does not directly apply to Go.
+// For XML specifications see http://help.catchsoftware.com/display/ET/JUnit+Format
+// or view the XSD included in this package as 'junit.xsd'
+
+// TestSuites represents a flat collection of jUnit test suites.
+type TestSuites struct {
+	XMLName xml.Name `xml:"testsuites"`
+
+	// Suites are the jUnit test suites held in this collection
+	Suites []*TestSuite `xml:"testsuite"`
```

Comment:
If we switch wholesale over to this approach, I would look forward to proper suite support. There's really nothing I could find that I can do in testgrid to reliably determine suite names other than hardcoding the ones we know about and hoping nobody adds more without us knowing beforehand. "."'s are in use in all kinds of test names.